### PR TITLE
Add codegeneration for stack.FunctionID

### PR DESCRIPTION
**Pull request by anatoly32322**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-
-
-

### CODE REVIEW

**Comment by anatoly32322** [comment]
Пока не до конца рабочий вариант:

- Не добавлено в CI
- Пару случаев FunctionID не видит
- Планирую упростить код, пока выглядит слишком запутанно из-за большого количества рекурсий

File: internal/stack/codegen/codegeneration.go
Comment (asmyasnikov) [inline internal/stack/codegen/codegeneration.go:1]: I think all internal tools must placed in internal/cmd package
What do you think about path internal/cmd/gstack ?

File: internal/stack/codegen/utils/utils.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
+package utils
+
+import (
+	"fmt"
+	"github.com/ydb-platform/ydb-go-sdk/v3/internal/stack"
```
Comment (asmyasnikov) [inline internal/stack/codegen/utils/utils.go:5]: Please fix imports ordering for pass gci linter checks

File: internal/stack/codegen/codegeneration.go
Comment (anatoly32322) [reply internal/stack/codegen/codegeneration.go:1]: Yeah, I'll move it to gstack

File: .github/workflows/check-codegen.yml [hunk 2]
```diff
@@ -32,6 +32,7 @@ jobs:
     - name: Build
       run: |
         go install ./internal/cmd/gtrace
+        go build ./internal/cmd/gstack
```
Comment (asmyasnikov) [inline .github/workflows/check-codegen.yml:35]:
```suggestion
        go install ./internal/cmd/gstack
```

File: .github/workflows/check-codegen.yml [hunk 3]
```diff
@@ -40,5 +41,10 @@ jobs:
         go generate ./trace
         go generate ./...
 
+    - name: Re-generate stack.FunctionID calls
+      run: |
+        ./gstack .
```
Comment (asmyasnikov) [inline .github/workflows/check-codegen.yml:46]:
```suggestion
        gstack .
```

File: .github/workflows/check-codegen.yml [hunk 4]
```diff
@@ -40,5 +41,10 @@ jobs:
         go generate ./trace
         go generate ./...
 
+    - name: Re-generate stack.FunctionID calls
+      run: |
+        gstack .
+        rm ./gstack
```
Comment (asmyasnikov) [inline .github/workflows/check-codegen.yml:47]:
```suggestion
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,7 +3,8 @@
 * Added query pool metrics
 * Fixed logic of query session pool
 * Changed initialization of internal driver clients to lazy
-* Disabled the logic of background grpc-connection parking 
+* Disabled the logic of background grpc-connection parking
+* Added codegarator for filling FunctionID with value from call stack
```
Comment (asmyasnikov) [inline CHANGELOG.md:7]:
```suggestion
* Added internal gstack codegen for filling `stack.FunctionID` with value from call stack
```

File: internal/cmd/gstack/utils/utils.go [hunk 6]
```diff
@@ -0,0 +1,134 @@
+package utils
+
+import (
+	"fmt"
+	"github.com/ydb-platform/ydb-go-sdk/v3/internal/version"
```
Comment (asmyasnikov) [inline internal/cmd/gstack/utils/utils.go:5]: gofumpt linter

**Pull request merged**